pkg/router/controller/status.go

```diff
@@ -234,14 +321,52 @@ func recordIngressCondition(route *routev1.Route, name, hostName string, conditi
 	return true, true, now.Time, ingress, nil
 }
 
-// findMostRecentIngress returns the name of the ingress status with the most recent Admitted condition transition time,
+// removeIngressCondition removes a matching status condition if it is found.
+// It returns whether the route was changed, the time it removed the condition, and
+// a pointer to the latest and original ingress records.
+func removeIngressCondition(route *routev1.Route, name string, condType routev1.RouteIngressConditionType) (changed bool, at time.Time, latest, original *routev1.RouteIngress) {
```

Comment:
Can you have more than 1 status? How does that work?

Reply:
Yes, you can. I added a unit test to for that in the rare scenario we'd ever encounter that.